src/modules/fancyzones/editor/FancyZonesEditor/Utils/FancyZonesEditorIO.cs

```diff
@@ -818,15 +818,16 @@ private bool SetCustomLayouts(List<CustomLayoutWrapper> customLayouts)
                     var zones = new List<Int32Rect>();
                     foreach (var zone in info.Zones)
                     {
-                        zones.Add(new Int32Rect { X = (int)zone.X, Y = (int)zone.Y, Width = (int)zone.Width, Height = (int)zone.Height });
+                        zones.Add(new Int32Rect { X = (int)zone.X, Y = (int)zone.Y, Width = Math.Max(zone.Width, 0), Height = Math.Max(zone.Height, 0) });
                     }
```

Comment:
Should we make the layout not valid instead of ignoring negative values?

Reply:
I'm not sure what's the best approach here. Zone width/height could be easily updated in the editor instead of malformed value.
But of course, making the layout invalid is the same approach as for grid layouts. 
@enricogior should we invalidate the canvas layout with malformed width/height?

Reply:
Yes, let's move all layouts that have malformed/invalid/inconsistent data.

Reply:
I do not have enough context here. But if we can recover without interrupting user experience(dialog box, noticeable lag) we should go with it.

Reply:
@mykhailopylyp 
malformed data is a bug, we know why we were causing it in 0.33 and we fixed it. If we detect any type of malformed data still present, we need to remove it. It's pointless to add code to workaround a bug that has been fixed (or trying to cover up for future possible bugs that cause malformed data).

Reply:
Then we should invalidate the canvas layout with malformed width/height if negative width/height is not expected.